### ISSUE TITLE
`prefer_final_parameters` should not flag params used in destructured patterns

### ISSUE THREAD

**Issue by srawlins** [dart-lang/linter#4156]
Child issue of https://github.com/dart-lang/linter/issues/3624 I suppose.

I think there should be a test case like:

```dart
void f(int two) {
  var list = [1, 2, 3];
  var [_, two, _] = list;
}
```

in which the lint rule would _not_ report that final should be preferred for `two`.

**Comment by pq** [comment]
See: https://dart-review.googlesource.com/c/sdk/+/289884

**Comment by pq** [comment]
Fixed as of #4188.
